contracts/aggregator/common/main.sol

```diff
@@ -0,0 +1,141 @@
+            routes.push(_routes[i]);
+        }
+    }
+
+    /**
+     * @dev Function to update existing routes.
+     * @notice Function to update existing routes and implementations.
+     * @param _routes routes to update.
+     * @param _impls implementations of their respective routes.
+     */
+    function updateRouteImplementations(uint256[] memory _routes, address[] memory _impls) public onlyOwner {
+        require(_routes.length == _impls.length, 'lengths-dont-match');
+        uint256 length = _routes.length;
+        for (uint256 i = 0; i < length; i++) {
+            routeToImpl[_routes[i]] = _impls[i];
+        }
+    }
+
+    /**
+     * @dev Function to change route status.
+     * @notice Function to enable and disable routes.
+     * @param _routes routes those status we want to change.
+     * @param _statuses new statuses.
+     */
+    function changeRouteStatus(uint256[] memory _routes, bool[] memory _statuses) public onlyOwner {
+        require(_routes.length == _statuses.length, 'lengths-dont-match');
+        uint256 length = _routes.length;
+        for (uint256 i = 0; i < length; i++) {
+            routeStatus[_routes[i]] = _statuses[i];
+        }
+    }
+
+    /**
+     * @dev Function to transfer fee to the treasury.
+     * @notice Function to transfer fee to the treasury. Will be called manually.
+     * @param _tokens token addresses for transferring fee to treasury.
+     * @param _to treasury address.
+     */
+    function transferFee(address[] memory _tokens, address _to) public onlyOwner {
+        uint256[] memory _amts = new uint256[](_tokens.length);
+        for (uint256 i = 0; i < _tokens.length; i++) {
+            IERC20 token_ = IERC20(_tokens[i]);
+            uint256 decimals_ = TokenInterface(_tokens[i]).decimals();
+            uint256 amtToSub_ = decimals_ == 18 ? 1e10 : decimals_ > 12 ? 10000 : decimals_ > 7 ? 100 : 10;
+            _amts[i] = token_.balanceOf(address(this)) > amtToSub_ ? (token_.balanceOf(address(this)) - amtToSub_) : 0;
+            if (_amts[i] > 0) token_.safeTransfer(_to, _amts[i]);
+        }
+        emit LogCollectRevenue(_to, _tokens, _amts);
+    }
+}
+
+contract FlashAggregator is AdminModule {
+
+    /**
+     * @dev Returns fee for the passed route in BPS.
+     * @notice Returns fee for the passed route in BPS. 1 BPS == 0.01%.
+     * @param _route route number for flashloan.
+     */
+    function calculateFeeBPS(uint256 _route) public view returns (uint256 BPS_) {
+        bytes memory _output = Address.functionStaticCall(routeToImpl[_route], msg.data, 'calculateFeeBPS-call-failed');
+        BPS_ = abi.decode(_output, (uint256));
+
+        if (BPS_ < InstaFeeBPS) {
+            BPS_ = InstaFeeBPS;
+        }
+    }
+
+    /**
+     * @dev Function to get the list of all routes.
+     * @notice Function to get the list of all routes.
+     */
+    function getRoutes() public view returns (uint256[] memory) {
+        return routes;
```

Comment:
Either this should return only those routes which have status `true` or we should use `getEnabledRoutes` function in the resolver

Reply:
Updated the `getEnabledRoutes` function and we're using this everywhere in resolver now.